bot/exts/moderation/stream.py

```diff
@@ -0,0 +1,161 @@
+                except discord.errors.NotFound:
+                    log.debug(
+                        f"Member {key} left the guild before we could schedule "
+                        "the revoking of their streaming permissions."
+                    )
+                    await self.task_cache.delete(key)
+                    continue
+                except discord.HTTPException:
+                    log.exception(f"Exception while trying to retrieve member {key} from Discord.")
+                    continue
+            revoke_time = Arrow.utcfromtimestamp(value)
+            log.debug(f"Scheduling {member} ({member.id}) to have streaming permission revoked at {revoke_time}")
+            self.scheduler.schedule_at(
+                revoke_time,
+                key,
+                self._revoke_streaming_permission(member)
+            )
+
+    @commands.command(aliases=("streaming",))
+    @commands.has_any_role(*STAFF_ROLES)
+    async def stream(self, ctx: commands.Context, member: discord.Member, duration: Expiry = None) -> None:
+        """
+        Temporarily grant streaming permissions to a member for a given duration.
+
+        A unit of time should be appended to the duration.
+        Units (∗case-sensitive):
+        \u2003`y` - years
+        \u2003`m` - months∗
+        \u2003`w` - weeks
+        \u2003`d` - days
+        \u2003`h` - hours
+        \u2003`M` - minutes∗
+        \u2003`s` - seconds
+
+        Alternatively, an ISO 8601 timestamp can be provided for the duration.
+        """
+        log.trace(f"Attempting to give temporary streaming permission to {member} ({member.id}).")
+        # If duration is none then calculate default duration
+        if duration is None:
+            duration = arrow.utcnow() + timedelta(minutes=VideoPermission.default_permission_duration)
+
+        # Check if the member already has streaming permission
+        already_allowed = any(Roles.video == role.id for role in member.roles)
+        if already_allowed:
+            await ctx.send(f"{Emojis.cross_mark} This member can already stream.")
+            log.debug(f"{member} ({member.id}) already has permission to stream.")
+            return
+
+        # Schedule task to remove streaming permission from Member and add it to task cache
+        self.scheduler.schedule_at(duration, member.id, self._revoke_streaming_permission(member))
+        await self.task_cache.set(member.id, Arrow.fromdatetime(duration).timestamp())
+
+        await member.add_roles(discord.Object(Roles.video), reason="Temporary streaming access granted")
+
+        # Convert here for nicer logging and output
+        revoke_time = format_infraction_with_duration(str(duration))
+        await ctx.send(f"{Emojis.check_mark} {member.mention} can now stream until {revoke_time}.")
+        log.debug(f"Successfully given {member} ({member.id}) permission to stream until {revoke_time}.")
```

Comment:
Missed this one, sorry.

```suggestion
        log.debug(f"Successfully gave {member} ({member.id}) permission to stream until {revoke_time}.")
```